src/sentry_sdk.cpp

```diff
@@ -0,0 +1,139 @@
+	if (p_user->get_id().is_empty()) {
+		// Take user ID from the runtime config or generate a new one if it's empty.
+		String user_id = get_user()->get_id();
+		if (user_id.is_empty()) {
+			user_id = sentry::uuid::make_uuid();
+		}
+		p_user->set_id(user_id);
+	}
+
+	// Save user in a runtime conf-file.
+	// TODO: Make saving optional?
+	runtime_config->set_user(p_user);
+	internal_sdk->set_user(p_user);
+}
+
+void SentrySDK::remove_user() {
+	internal_sdk->remove_user();
+}
+
+void SentrySDK::set_context(const godot::String &p_key, const godot::Dictionary &p_value) {
+	ERR_FAIL_COND_MSG(p_key.is_empty(), "Sentry: Can't set context with an empty key.");
+	internal_sdk->set_context(p_key, p_value);
+}
+
+void SentrySDK::_init_contexts() {
+	internal_sdk->set_context("device", sentry::contexts::make_device_context(runtime_config));
+	internal_sdk->set_context("app", sentry::contexts::make_app_context());
+	internal_sdk->set_context("gpu", sentry::contexts::make_gpu_context());
+	internal_sdk->set_context("culture", sentry::contexts::make_culture_context());
+
+	// Custom contexts.
+	internal_sdk->set_context("display", sentry::contexts::make_display_context());
+	internal_sdk->set_context("godot_engine", sentry::contexts::make_godot_engine_context());
+	internal_sdk->set_context("environment", sentry::contexts::make_environment_context());
+}
+
+void SentrySDK::_bind_methods() {
+	BIND_ENUM_CONSTANT(LEVEL_DEBUG);
+	BIND_ENUM_CONSTANT(LEVEL_INFO);
+	BIND_ENUM_CONSTANT(LEVEL_WARNING);
+	BIND_ENUM_CONSTANT(LEVEL_ERROR);
+	BIND_ENUM_CONSTANT(LEVEL_FATAL);
+
+	ClassDB::bind_method(D_METHOD("capture_message", "message", "level", "logger"), &SentrySDK::capture_message, DEFVAL(LEVEL_INFO), DEFVAL(""));
+	ClassDB::bind_method(D_METHOD("add_breadcrumb", "message", "category", "level", "type", "data"), &SentrySDK::add_breadcrumb, DEFVAL(LEVEL_INFO), DEFVAL("default"), DEFVAL(Dictionary()));
+	ClassDB::bind_method(D_METHOD("get_last_event_id"), &SentrySDK::get_last_event_id);
+	ClassDB::bind_method(D_METHOD("set_context", "key", "value"), &SentrySDK::set_context);
+	ClassDB::bind_method(D_METHOD("set_tag", "key", "value"), &SentrySDK::set_tag);
+	ClassDB::bind_method(D_METHOD("remove_tag", "key"), &SentrySDK::remove_tag);
+	ClassDB::bind_method(D_METHOD("set_user", "user"), &SentrySDK::set_user);
+	ClassDB::bind_method(D_METHOD("get_user"), &SentrySDK::get_user);
+	ClassDB::bind_method(D_METHOD("remove_user"), &SentrySDK::remove_user);
+}
+
+SentrySDK::SentrySDK() {
+	ERR_FAIL_NULL(OS::get_singleton());
+	ERR_FAIL_NULL(SentryOptions::get_singleton());
+
+	singleton = this;
+
+	// Load the runtime configuration from the user's data directory.
+	runtime_config.instantiate();
+	runtime_config->load_file(OS::get_singleton()->get_user_data_dir() + "/sentry.dat");
+
+#ifdef NATIVE_SDK
+	internal_sdk = std::make_shared<NativeSDK>();
+	enabled = true;
+#else
+	// Unsupported platform
+	internal_sdk = std::make_shared<DisabledSDK>();
```

Comment:
would be nice to debug log here too: This is an unsupported platform. This is what you can expect: ...

Reply:
Done

Reply:
Please mark "Resolve Conversation" so it's easier to know what requires my input

![image](https://github.com/user-attachments/assets/3626884d-5f90-4207-8781-57c1f0f4ae1c)